langroid/agent/task.py

```diff
@@ -582,16 +587,44 @@ def reset_all_sub_tasks(self) -> None:
         for t in self.sub_tasks:
             t.reset_all_sub_tasks()
 
+    @overload
+    def run(  # noqa
+        self,
+        msg: Any | None = None,
+        *,
+        turns: int = -1,
+        caller: None | Task = None,
+        max_cost: float = 0,
+        max_tokens: int = 0,
+        session_id: str = "",
+        allow_restart: bool = True,
+    ) -> Optional[ChatDocument]: ...  # noqa
+
+    @overload
+    def run(  # noqa
+        self,
+        msg: Any | None = None,
+        *,
+        turns: int = -1,
+        caller: None | Task = None,
+        max_cost: float = 0,
+        max_tokens: int = 0,
+        session_id: str = "",
+        allow_restart: bool = True,
+        return_type: Type[T],
+    ) -> Optional[T]: ...  # noqa
+
     def run(
         self,
-        msg: Optional[str | ChatDocument] = None,
+        msg: Any | None = None,
```

Comment:
Here, and in `run_async`, this can just be `Any`

Reply:
Pushed this change, but the original may be more readable, let me know what you think